custom_components/zaptec/api.py

```diff
@@ -295,6 +297,7 @@ async def _stream(self, cb=None):
     async def cancel_stream(self):
         if self._stream_task is not None:
             try:
+                await self._receiver.close()
```

Comment:
Can try do it more defensively? Check that self._receiver is not and check that is has a close method

Reply:
Yes, that is probably a good idea. When the stream task is waiting on `async for msg in receiver:`, it will hang when the stream task is canceled and awaited on. It must relinguish this loop and calling `self._receiver.close()` worked, albeit a bit brute force.

Reply:
Updated to more defensive coding. Added try...finally to ensure the receiver object is only valid during reception and check that its not empty before calling `close()`.

I didn't add a check for close thou. Isn't it ok to assume that the servicebus client receiver indeed has a `close()` method as its docs claim?